### PR TITLE
Refine tier 3 platform support definition

### PR DESCRIPTION
**Pull request by hasheddan**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

Updates tier 3 platform support definition to require demonstrated
ability to target the platform and documentation available for doing so.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

#### Which issue(s) this PR fixes:

xref #1337 

#### Special notes for your reviewer:

Updates based on feedback in [SIG Arch meeting](https://youtu.be/CMDqZJy1-gk).

/assign @saschagrunert @justaugustus 
/cc @dims

### CODE REVIEW

File: release-engineering/platforms.md [hunk 1]
```diff
@@ -43,12 +43,14 @@ platform.
...
-for, but which are not built or tested automatically. This means the binary
-artifacts may not work as intended.
+Tier 3 platforms are those that have been demonstrated to work and have a
+documented build process that is available to anyone in the community. There is
+no guarantee that builds will continue to work, and the platform may be dropped
+if documentation is not maintained.
 
 - Official builds are not available.
 - Automated testing is not setup.
+- Documentation on how to build for the platform is available.
```
Comment (hasheddan) [inline release-engineering/platforms.md:53]: I am trying to decide if this documentation belongs in sig-release/release-engineering or in https://github.com/kubernetes/community/tree/master/contributors/devel/sig-architecture. I would like to keep as much of this information located together as possible, but we already have the beginning of guide in k/community. Ultimately, it seems like this primarily falls under release-engineering in terms of shepherding people through the process, and we also need to make sure release-engineering has knowledge of how to build all platforms, so I am somewhat inclined to move the guide here, point folks to it, and have Tier 3 build docs also live here.

Thoughts? @dims @saschagrunert

Comment (dims) [reply release-engineering/platforms.md:53]: release-engineering sounds good to me!

Comment (saschagrunert) [reply release-engineering/platforms.md:53]: I agree this should live here.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**